src/scaffold/components/Card.tsx

```diff
@@ -0,0 +1,69 @@
+import * as React from 'react';
+import selectedItemIcon from '../../../resources/selected-item.png';
+import { Uri } from 'vscode';
+import { useStore } from '../store';
+
+interface CardProps {
+  type: string;
+  iconUri?: Uri;
+  key: string;
+  title: string;
+  description: string;
+  shortName?: string;
+  author?: string;
+  handleOnClick: () => void;
+}
+
+const Card: React.FC<CardProps> = ({
+  iconUri,
+  title,
+  description,
+  type,
+  shortName,
+  author,
+  handleOnClick,
+}) => {
+  const { state } = useStore();
+  const options = state.options;
+  const [allClasses, setAllClasses] = React.useState(`card ${type}`);
+
+  function getRef(iconUri: Uri) {
+    const { scheme, authority, path } = iconUri;
+    return `${scheme}://${authority}${path}`;
+  }
+
+  React.useEffect(() => {
+    switch (type) {
+      case 'type':
+        options.repoType === title ? setAllClasses(`card ${type} selectedCard`) : setAllClasses(`card ${type}`);
+        break;
+
+      case 'bundler':
+        options.bundler === title ? setAllClasses(`card ${type} selectedCard`) : setAllClasses(`card ${type}`);
+        break;
+
+      case 'template':
+        options.template === title ? setAllClasses(`card ${type} selectedCard`) : setAllClasses(`card ${type}`);
+        break;
+    }
+  }, [state.options]);
```

Comment:
Should only depend on `type` and then either `options.repoType`, `options.bundler` or `options.template`. Right now any change of options will re-run the effect.